### PR TITLE
Expose JsonSerializerSettings in ApiClient

### LINKED ISSUE #4241: [REQ] [csharp-netcore] Custom JsonSerializer needed
### Is your feature request related to a problem? Please describe.

Lib: csharp-netcore

I want to do a "Patch" request to update some entries.
For this reason its needed that "null" values are not serialized. If a property is not serialized it will be ignored from the service.

## Describe the solution you'd like

To solve this i want to add a custom Json serializer. Inside this I'm able to say that null values will be ignored.

example how it worket after i changed the generated code:

```
        private RestRequest newRequest(
            HttpMethod method,
            String path,
            RequestOptions options,
            IReadableConfiguration configuration)
        {

....
            RestRequest request = new RestRequest(Method(method))
            {
                Resource = path,
                JsonSerializer = new CustomJsonCodec(new JsonSerializerSettings
                {
                  NullValueHandling = NullValueHandling.Ignore
                },configuration)
            };
```
But we want to make it more generic :-)
The creation of this request is wrapped by the generated client. And there is actual no way to pass a custom Json serializer. I think its a good idea to pass a custon serializer together wirh the configuration. With this the client can configure what he want.

## Describe alternatives you've considered

Its also possible to add some configurations like `NullValueHandling = NullValueHandling.Ignore` directly to the config. But i think thats not so suitable.

### PR DESCRIPTION
**Pull request by davidyee**
In this PR, I expose the JsonSerializerSettings in the ApiClient so that serialization customizations can be done. For example, in my specific use-case, I wanted to add additional Converters to change the way JSON was being deserialized back into the csharp classes.

Fixes #4241

I reviewed the comments made on this related PR #4466 and incorporated the feedback into this PR.

@mandrean (2017/08) @frankyjuang (2019/09) @shibayan (2020/02)

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/csharp-netcore/ApiClient.mustache [hunk 1]
```diff
@@ -147,6 +147,19 @@ namespace {{packageName}}.Client
     {
         private readonly String _baseUrl;
 
+        public JsonSerializerSettings SerializerSettings { get; set; } = new JsonSerializerSettings
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/csharp-netcore/ApiClient.mustache:150]: @davidyee can you please add XML style comments for `SerializerSettings`?

**Comment by wing328** [comment]
@davidyee thanks for the PR. I wonder if you can add a new `ApiClientTests.cs` file to test `JsonSerializerSettings`

A good reference is `ConfigurationTests.cs`

Ref: https://github.com/OpenAPITools/openapi-generator/blob/master/samples/client/petstore/csharp-netcore/OpenAPIClient/src/Org.OpenAPITools.Test/ConfigurationTests.cs

**Comment by wing328** [comment]
FYI. I've filed https://github.com/OpenAPITools/openapi-generator/pull/7583 to default `OverrideSpecifiedNames` to false. Let me know if you've any concerns.

**Comment by davidyee** [comment]
@wing328 Thanks for the feedback and suggestions. I've included revisions to my PR in my latest commits.

**Comment by wing328** [comment]
@davidyee thanks for the PR, which has been merged into master.

**Pull request merged**